persistence-zookeeper/src/main/scala/com/mesosphere/usi/storage/zookeeper/PodRecordRepository.scala

```diff
@@ -0,0 +1,48 @@
+package com.mesosphere.usi.storage.zookeeper
+
+import akka.util.ByteString
+import com.mesosphere.usi.core.models.{PodId, PodRecord}
+import com.mesosphere.usi.repository.{PodRecordRepository => PodRecordRepositoryInterface}
+import com.mesosphere.usi.storage.zookeeper.PersistenceStore.Node
+import org.apache.zookeeper.KeeperException.NoNodeException
+
+import scala.async.Async.{async, await}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+class PodRecordRepository(val store: PersistenceStore) extends PodRecordRepositoryInterface {
+
+  override def create(record: PodRecord): Future[PodId] = async {
+    val data = record.toString
+    val node = Node(record.podId.value, ByteString(data))
+    await(store.create(node))
+    record.podId
+    // TODO: RecordAlreadyExistsException
```

Comment:
So yeah. The interface states that the future should fail if the record already exists. How are we going to do that? Should we use a transaction here? The `TemplateRepository` does not have this behaviour as far as I know. It uses the `PathTrie` which synchronizes requests.

Reply:
I think we must use a transaction here.

Reply:
> The `TemplateRepository` does not have this behavior as far as I know.

it does. creating a node with an existing id/path is invalid and should fail.

Reply:
There is only one `create` request here - so what should a transaction do here?

Reply:
First. You are on vacation so why should I listen to you 😉 

Second. Show me the code or it did not happen. The `TemplateRepository.create` does not check if the node exists. Maybe this is something in Curator that throws an exception. I'm going to test that today.

Reply:
> Maybe this is something in Curator that throws an exception.

It's ZK ;) A `NodeExistsException is thrown if you try to create an existing node. A `NoNodeException` is thrown if you try to update a **non-existing** node. Both makes total sense.

Reply:
Found it. Thanks